### PR TITLE
docs: document explain api (close #870)

### LINKED ISSUE #870: document the explain api

### PR DESCRIPTION
**Pull request by marionschleifer**
### Description

Document the `/v1/graphql/explain` endpoint.

### Affected components
- Docs

### CODE REVIEW

**Comment by marionschleifer** [comment]
@0x777 is there another expression we can use for the object that is returned from the API, other than "Postgres plan"? It's a little confusing because one value inside the returned JSON object is `"plan"`.

@rikinsk I'm not sure if we should move this section under the GraphQL API, since the Explain API is part of the GraphQL API. The disadvantage of that would be that it's probably harder to find and to search for. Let me know your thoughts.

**Comment by rikinsk** [comment]
@marionschleifer The explain API should be its own thing, I dont think it belongs under the GraphQL API section as this is not user-facing like queries and mutations and will be confusing there.

**Comment by marionschleifer** [comment]
@0x777 @rikinsk please add your reviews 🙂

File: docs/graphql/manual/api-reference/index.rst [hunk 1]
```diff
@@ -15,23 +15,25 @@ API Reference
...
++-----------------+-----------------------------------------+------------------+
+| Version         | :ref:`/v1/version <version_api>`        | Public           |
++-----------------+-----------------------------------------+------------------+
+| Health          | :ref:`/healthz <health_api>`            | Public           |
++-----------------+-----------------------------------------+------------------+
+| PG Dump         | :ref:`/v1alpha1/pg_dump <pg_dump_api>`  | Admin only       |
++-----------------+-----------------------------------------+------------------+
+| Config          | :ref:`/v1alpha1/config <config_api>`    | Admin only       |
++-----------------+-----------------------------------------+------------------+
+| Explain         | :ref:`/v1/graphql/explain <explain_api>`| Permission rules |
```
Comment (0x777) [inline docs/graphql/manual/api-reference/index.rst]: Explain endpoints are only available for admin.

File: docs/graphql/manual/api-reference/explain.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            "              ->  Seq Scan on authors  (cost=0.00..11.83 rows=610 width=32)",
+            "              SubPlan 1",
+            "                ->  Result  (cost=0.00..0.00 rows=1 width=32)"
+        ]
+    }
+
+Disabling Explain API
+---------------------
+
+The Explain API is part of the :doc:`GraphQL API <graphql-api/index>` and can only be disabled by disabling the same.
```
Comment (0x777) [inline docs/graphql/manual/api-reference/explain.rst]: It is now part of `metadata` group, so can be disable by disabling `metadata` api.

**Pull request merged**